core/src/main/scala/kafka/admin/LeaderElectionCommand.scala

```diff
@@ -175,6 +178,60 @@ object LeaderElectionCommand extends Logging {
       throw rootException
     }
   }
+
+  private[this] def validate(commandOptions: LeaderElectionCommandOptions): Unit = {
+    // required options: --bootstrap-server and --election-type
+    var missingOptions = List.empty[String]
+    if (!commandOptions.options.has(commandOptions.bootstrapServer)) {
+      missingOptions = commandOptions.bootstrapServer.options().get(0) :: missingOptions
+    }
+
+    if (!commandOptions.options.has(commandOptions.electionType)) {
+      missingOptions = commandOptions.electionType.options().get(0) :: missingOptions
+    }
+
+    if (missingOptions.nonEmpty) {
+      throw new AdminCommandFailedException(s"Missing required option(s): ${missingOptions.mkString(", ")}")
+    }
+
+    // One and only one is required: --topic, --all-topic-partitions or --path-to-json-file
+    (
+      commandOptions.options.has(commandOptions.topic),
```

Comment:
I guess another way to do this might be to use something like this:
```scala
val mutuallyExclusiveOptions = Seq(
  commandOption.topic, 
  commandOptions.allTopicPartitions, 
  commandOptions.pathToJsonFile)

mutuallyExclusiveOptions.count(commandOptions.options.has) match {
  case 0 =>
  case 1 =>
  case _ =>
}
```

Reply:
Much better. Fixed.